components/dashboard/src/components/DropDown2.tsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import React, { FunctionComponent, RefObject, useCallback, useEffect, useMemo, useRef, useState } from "react";
+import Arrow from "./Arrow";
+
+export interface DropDown2Element {
+    id: string;
+    element: JSX.Element;
+    isSelectable?: boolean;
+}
+
+export interface DropDown2Props {
+    getElements: (searchString: string) => DropDown2Element[];
+    searchPlaceholder?: string;
+    disableSearch?: boolean;
+    expanded?: boolean;
+    onSelectionChange: (id: string) => void;
+}
+
+export const DropDown2: FunctionComponent<DropDown2Props> = (props) => {
+    const [showDropDown, setShowDropDown] = useState<boolean>(!!props.expanded);
+    const nodeRef: RefObject<HTMLDivElement> = useRef(null);
+    const onSelected = useCallback(
+        (elementId: string) => {
+            props.onSelectionChange(elementId);
+            setShowDropDown(false);
+        },
+        [props],
+    );
+    const [search, setSearch] = useState<string>("");
+    const filteredOptions = useMemo(() => props.getElements(search), [props, search]);
+    const [selectedElementTemp, setSelectedElementTemp] = useState<string | undefined>(filteredOptions[0]?.id);
+
+    // reset search when the drop down is expanded or closed
+    useEffect(() => {
+        setSearch("");
+        if (showDropDown && selectedElementTemp) {
+            document.getElementById(selectedElementTemp)?.scrollIntoView({ behavior: "smooth", block: "nearest" });
```

Comment:
This is a non-blocking comment, but something to maybe think about as a way to improve the UX of this component. While this behavior is nice in that it helps show a selected option - it feels a bit disorienting to watch a long list scroll when you open the dropdown. It also adds an additional step if you just wanted to type to filter after you expanded the list (now the search input is not focused and out of view). 

A more common UX pattern for interfaces like this I think that can fit well would be to have the selected component always show up on top of the list of options, and then the search input can stay focused when the dropdown opens, making it easy to type, and you also see the selected option.

Another option could be having the search input not in the scrollable container, and stay pinned so that only the options scroll. Either way, I don't think this is a blocker, just some suggestions to consider if we want to improve it a bit.